pallets/registrar/rpc/runtime-api/src/lib.rs

```diff
@@ -36,3 +36,19 @@ sp_api::decl_runtime_apis! {
         fn boot_nodes(para_id: ParaId) -> Vec<Vec<u8>>;
     }
 }
+
+sp_api::decl_runtime_apis! {
+    pub trait OnDemandBlockProductionApi<ParaId, Slot> where
+        ParaId: parity_scale_codec::Codec,
+        Slot: parity_scale_codec::Codec,
+    {
+        /// Inclusive range of slots during which collators can propose the next block.
+        ///
+        /// # Returns
+        ///
+        /// Range `Some((start, end))`, where the condition for the slot to be valid is
+        /// `(slot - parent_slot) >= start && (slot - parent_slot) <= end`.
+        /// `None` if the `para_id` is not a parathread.
+        fn next_slot_range_inclusive(para_id: ParaId) -> Option<(Slot, Slot)>;
```

Comment:
Shall we remove the end? I guess this is related to the max parameter, which we said we should probably skip?

Reply:
Yes, I think I will return only the `start`. But I need to experiment a bit on runtime apis to see how easy it will be to change the method signature in the future.

Reply:
So regarding runtime api versioning, it looks like changing a method signature is not possible. The only way to change it is to create a new method, which will return an error if called using an older runtime.

So I created a method that returns an `Option<start>`, which should cover all the possible user cases. Returning `None` will make clients create blocks immediately, and returning `Some(u32::MAX)` can be used to prevent outdated clients from creating blocks.